homeassistant/components/mobile_app/webhook.py

```diff
@@ -0,0 +1,174 @@
+                         webhook_id: str, request: Request) -> Response:
+    """Handle webhook callback."""
+    if webhook_id in hass.data[DOMAIN][ATTR_DELETED_IDS]:
+        return Response(status=410)
+
+    device = hass.data[DOMAIN][ATTR_REGISTRATIONS][webhook_id]
+
+    try:
+        req_data = await request.json()
+    except ValueError:
+        _LOGGER.warning('Received invalid JSON from mobile_app')
+        return json_response([], status=HTTP_BAD_REQUEST)
+
+    try:
+        req_data = WEBHOOK_PAYLOAD_SCHEMA(req_data)
+    except vol.Invalid as ex:
+        err = vol.humanize.humanize_error(req_data, ex)
+        _LOGGER.error('Received invalid webhook payload: %s', err)
+        return empty_okay_response()
+
+    headers = {}
+
+    via_cloud = is_cloudhook_request(request)
+
+    logged_in = async_is_logged_in(hass)
+
+    if via_cloud is False and logged_in is True:
+        cloud_id = device.get(CONF_CLOUDHOOK_ID)
+        cloud_url = device.get(CONF_CLOUDHOOK_URL)
+        if cloud_url is None:
+            try:
+                hook = await async_create_cloudhook(hass, webhook_id)
+                cloud_id = hook[CONF_CLOUDHOOK_ID]
+                cloud_url = hook[CONF_CLOUDHOOK_URL]
+            except CloudNotAvailable:
+                _LOGGER.error("Error creating cloudhook during local call")
+
+        headers[HTTP_X_CLOUD_HOOK_ID] = cloud_id
+        headers[HTTP_X_CLOUD_HOOK_URL] = cloud_url
+
+    webhook_type = req_data[ATTR_WEBHOOK_TYPE]
+
+    webhook_payload = req_data.get(ATTR_WEBHOOK_DATA, {})
+
+    if req_data[ATTR_WEBHOOK_ENCRYPTED]:
+        enc_data = req_data[ATTR_WEBHOOK_ENCRYPTED_DATA]
+        webhook_payload = _decrypt_payload(device[CONF_SECRET], enc_data)
+
+    if (webhook_type not in WEBHOOK_TYPES and
+            device[ATTR_APP_ID] in INTEGRATIONS_MAP):
+        # Unknown webhook type, check if there's a platform
+        platform_name = INTEGRATIONS_MAP[device[ATTR_APP_ID]]
+
+        platform = get_platform(hass, DOMAIN, platform_name)
+        return await platform.async_handle_webhook_message(hass, device,
+                                                           webhook_type,
+                                                           webhook_payload)
+
+    try:
+        data = WEBHOOK_SCHEMAS[webhook_type](webhook_payload)
+    except vol.Invalid as ex:
+        err = vol.humanize.humanize_error(webhook_payload, ex)
+        _LOGGER.error('Received invalid webhook payload: %s', err)
+        return empty_okay_response(headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_CALL_SERVICE:
+        try:
+            await hass.services.async_call(data[ATTR_DOMAIN],
+                                           data[ATTR_SERVICE],
+                                           data[ATTR_SERVICE_DATA],
+                                           blocking=True,
+                                           context=device_context(device))
+        except (vol.Invalid, ServiceNotFound):
```

Comment:
You should also catch bare Exception, and if so `_LOGGER.exception` it, and include the device ID

Reply:
(pylint disable the bare except)

Reply:
I assume you meant device name since we killed device ID? Or did you mean webhook ID?